apis/dns/v1alpha1/dns_types.go

```diff
@@ -0,0 +1,397 @@
+
+	// Metadata - The metadata attached to the record set
+	// +optional
+	Metadata map[string]string `json:"metadata"`
+
+	// TTL - The TTL (time-to-live) of the records in the record set.
+	// +required
+	TTL *int `json:"ttl"`
+
+	// TargetResource - A reference to an azure resource from where the dns resource value is taken.
+	// +optional
+	TargetResource *SubResource `json:"targetResource,omitempty"`
+
+	// RecordType enumerates the values for record type.
+	// +required
+	RecordType RecordType `json:"recordType"`
+
+	// ARecords - The list of A records in the record set.
+	// +optional
+	ARecords []ARecord `json:"aRecords,omitempty"`
+
+	// AAAARecords - The list of AAAA records in the record set.
+	// +optional
+	AAAARecords []AAAARecord `json:"aaaaRecords,omitempty"`
+
+	// MXRecords - The list of MX records in the record set.
+	// +optional
+	MXRecords []MXRecord `json:"mxRecords,omitempty"`
+
+	// NSRecords - The list of NS records in the record set.
+	// +optional
+	NSRecords []NSRecord `json:"nsRecords,omitempty"`
+
+	// PTRRecords - The list of PTR records in the record set.
+	// +optional
+	PTRRecords []PTRRecord `json:"ptrRecords,omitempty"`
+
+	// SRVRecords - The list of SRV records in the record set.
+	// +optional
+	SRVRecords []SRVRecord `json:"srvRecords,omitempty"`
+
+	// TXTRecords - The list of TXT records in the record set.
+	// +optional
+	TXTRecords []TXTRecord `json:"txtRecords,omitempty"`
+
+	// CNAMERecord - The CNAME record in the  record set.
+	// +optional
+	CNAMERecord CNAMERecord `json:"cnameRecord,omitempty"`
+
+	// SOARecord - The SOA record in the record set.
+	// +optional
+	SOARecord SOARecord `json:"soaRecord,omitempty"`
+
+	// CAARecords - The list of CAA records in the record set.
+	// +optional
+	CAARecords []CAARecord `json:"caaRecords,omitempty"`
+}
+
+// RecordSetObservation define the actual state of an Azure DNS RecordSet.
+type RecordSetObservation struct {
+	// ID - Resource ID
+	ID string `json:"id,omitempty"`
+
+	// Etag - The etag of the zone.
+	Etag string `json:"etag,omitempty"`
+
+	// Name - The name of the zone.
+	Name string `json:"name,omitempty"`
+
+	// Type - Resource type.
+	Type string `json:"type,omitempty"`
+
+	// FQDN - Fully qualified domain name of the record set.
+	FQDN string `json:"fqdn,omitempty"`
+	// ProvisioningState -provisioning State of the record set.
+	ProvisioningState string `json:"provisioningState,omitempty"`
+}
+
+// ARecord an A record.
+type ARecord struct {
+	// IPV4Address - The IPv4 address of this A record.
+	IPV4Address string `json:"ipv4Address,omitempty"`
+}
+
+// AAAARecord an AAAA record.
+type AAAARecord struct {
+	// IPV6Address - The IPv6 address of this AAAA record.
+	IPV6Address string `json:"ipv6Address,omitempty"`
+}
+
+// CAARecord a CAA record.
+type CAARecord struct {
+	// Flags - The flags for this CAA record as an integer between 0 and 255.
+	Flags int `json:"flags,omitempty"`
```

Comment:
If this is optional:
```suggestion
        // +optional
	Flags *int `json:"flags,omitempty"`
```
Could you please check other fields as well?

Reply:
I checked these fields. For all record structs, the fields are not required in Azure SDK. So they were marked as optional and also converted to pointer.